modules/lightning_features/lightning_api/modules/api_test/tests/src/Functional/ApiTestBase.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\Tests\api_test\Functional;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Url;
+use Drupal\Tests\BrowserTestBase;
+use Psr\Http\Message\ResponseInterface;
+
+abstract class ApiTestBase extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $profile = 'lightning_headless';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = ['api_test'];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Generate and store keys for use by OAuth.
+    $this->generateKeys();
+  }
+
+  /**
+   * Gets a token from the oauth endpoint for use in requests that require
+   * authorization.
+   *
+   * @param array $client_options
+   *   An array of options to use when requesting the token.
+   *
+   * @return string
+   *   The OAuth2 password grant access token from the API.
+   */
+  protected function getToken($client_options) {
+    $client = \Drupal::httpClient();
```

Comment:
Let us use $this->container->get('http_client')

Reply:
✔️